### PR TITLE
Update Match Score and State

### PR DESCRIPTION
**Pull request by Malaber**
[TURNIERE-106](https://jiratinf16b4.it.dh-karlsruhe.de:8443/browse/TURNIERE-106)

### CODE REVIEW

File: app/services/playoff_stage_service.rb [hunk 1]
```diff
@@ -65,6 +66,7 @@ def self.calculate_required_stage_count(number_of_teams)
       0
     else
       # black voodoo magic
+      # h
```
Comment (Thor77) [inline app/services/playoff_stage_service.rb]: remove your typo pls (:

File: app/interactors/update_match_score.rb [hunk 2]
```diff
@@ -0,0 +1,16 @@
+# frozen_string_literal: true
+
+class UpdateMatchScore
+  include Interactor
+
+  def call
+    match_score = context.match_score
+    match_score_params = context.match_score_params
+    if match_score.update(match_score_params)
+      match_score.match.state = 'in_progress'
```
Comment (Thor77) [inline app/interactors/update_match_score.rb]: Why are you setting the state to in_progress all the time?
You probably want to call your evaluate_status method instead...

File: app/models/match.rb [hunk 3]
```diff
@@ -11,22 +11,24 @@ class Match < ApplicationRecord
 
   validate :stage_xor_group
 
+  def evaluate_match_result
+    if match_scores.first == match_scores.second
```
Comment (Thor77) [inline app/models/match.rb]: You're comparing objects here.. they will (probably) never be the same, therefore this method will return wrong results (a spec for this method should catch this)
You should use `match_scores.first/second.points` instead.

**Review by Thor77 — CHANGES_REQUESTED** [review]
I found some issues..
Additionally please add a spec for `Match.evaluate_match_result` (in the Match model spec-file)

File: app/interactors/update_match_score.rb (hunk 2 above)
Comment (Thor77) [reply app/interactors/update_match_score.rb]: As discussed we will take care of updating the match state in a separate ticket/pr, therefore remove this line pls.

File: app/models/match.rb (hunk 3 above)
Comment (Thor77) [reply app/models/match.rb]: As discusssed please remove this method from the PR and add it in the one adding a match finish method.

File: app/services/playoff_stage_service.rb (hunk 1 above)
Comment (Malaber) [reply app/services/playoff_stage_service.rb]: did that

**Comment by Thor77** [comment]
I don't really see the point when it's basically a huge wrapper around the `Match.update` method..
When you set the state as well there was at least **some** logic, but now it's just a 83-line replacement for one line...

**Pull request closed without merging**